### PR TITLE
#246 - Added RxLock

### LINKED ISSUE #246: Reactive lock to work better with RxStorage
ASTO library provides `RxStorage` wrapper to use instead of `Storage` and it is used in adapters. When using with `Lock` it is needed to adapt this kind of storage manually. It would be nice to have such adapter ready in library

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #246 
Added `RxLock`

### CODE REVIEW

File: src/main/java/com/artipie/asto/lock/RxLock.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import hu.akarnokd.rxjava2.interop.CompletableInterop;
+import io.reactivex.Completable;
+
+/**
+ * Asynchronous lock that might be successfully obtained by one thread only at a time.
+ *
+ * @since 0.27
+ */
+public interface RxLock {
```
Comment (g4s8) [inline src/main/java/com/artipie/asto/lock/RxLock.java:34]: @olegmoz do we really need an interface here? This class seems to be a wrapper for `Lock` to use it for RX operations, similar to `BlockingStorage` for `Storage`. I'd not create new interfaces if we won't inject it as a dependency or pass as method parameter ever.

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olegmoz please check my comment

File: src/main/java/com/artipie/asto/lock/RxLock.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/asto/lock/RxLock.java]: @g4s8 yes, I think you are right. No point in making it an interface

**Comment by g4s8** [comment]
@olegmoz please check CI build logs, some tests are failing

**Comment by olegmoz** [comment]
@g4s8 that was a test crash due to timeout, now all the checks pass

**Review by g4s8 — APPROVED** [review]
@olegmoz thanks

**Comment by 0crat** [comment]
Job `gh:artipie/asto#247` is not assigned, can't get performer

**Pull request merged**